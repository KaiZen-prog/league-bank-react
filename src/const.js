// Common
export const DESKTOP_MIN_WIDTH = 1024;
export const DIGIT_SPACE = 3;
export const PHONE_LENGTH = 18;
export const RATING_SCALE_MULTIPLIER = 20;

export const KeyCode = {
  ESC: 27,
};

export const AppRoute = {
  MAIN: '',
  CONVERTER: '/converter',
};
// ---------------------------------

// Converter
export const FLOAT_COEFFICIENT = 100;
export const MAX_HISTORY_LENGTH = 10;

export const Currencies = ['RUB', 'USD', 'EUR', 'GBP', 'CNY'];

export const FormFields = {
  INPUT: 'currencyInput',
  OUTPUT: 'currencyOutput',
};

export const ConverterInputParams = {
  maxValue: 1000000,

  input: {
    id: 'currency-input',
    name: 'currencyInput',
  },

  output: {
    id: 'currency-output',
    name: 'currencyOutput',
  }
};
// ---------------------------------

// Calculator
export const REQUIRED_INCOME = 45;
export const QUANTITY_MONTH = 12;

export const CalculatorSteps = {
  params: 'params',
  request: 'request',
};

export const CreditPurpose = {
  none: {
    name: 'Выберите цель кредита',
    type: '',
  },
  mortgage: {
    name: 'Ипотечное кредитование',
    type: 'mortgage',
  },
  car: {
    name: 'Автомобильное кредитование',
    type: 'car',
  },
};

export const InputFields = {
  cost: 'cost',
  initialFee: 'initialFee',
  term: 'term',
};

export const LabelTypes = {
  car: 'car',
};

export const OfferTypes = {
  refusal: 'refusal',
};

export const SubmitButtonTypes = {
  preorder: 'preorder',
  request: 'request',
};

export const InputIconsTypes = {
  minus: 'minus',
  plus: 'plus',
};

export const InputTypes = {
  email: 'email',
  fullName: 'fullName',
  initialFee: 'initialFee',
  phone: 'phone',
  show: 'show',
  term: 'term',
  userInfo: 'userInfo',
};

export const MortgageParams = {
  type: 'mortgage',
  minCost: 1200000,
  maxCost: 25000000,
  step: 100000,
  minInitialFee: 10,
  minTerm: 5,
  maxTerm: 30,
  minCreditAmount: 500000,
  maternalCapitalValue: 470000,
  percent: {
    defaultPercent: 9.4,
    specialPercent: 8.5,
    amountForSpecialPercent: 15,
  },
};

export const CarParams = {
  type: 'car',
  minCost: 500000,
  maxCost: 5000000,
  step: 50000,
  minInitialFee: 20,
  minTerm: 1,
  maxTerm: 5,
  minCreditAmount: 200000,
  additionalToCar: {
    casco: 'casco',
    lifeInsurance: 'lifeInsurance',
  },
  percent: {
    defaultPercent: 16,
    specialPercent: 15,
    amountForSpecialPercent: 2000000,
    oneAddition: 8.5,
    allAdditions: 3.5,
  },
};
// ---------------------------------

// Slider
export const Sliders = {
  main: {
    name: 'main-slider',
    slides: {
      credit: 'credit',
      promo: 'promo',
      offices: 'offices',
    },
  },

  services: {
    name: 'services-slider',
    slides: {
      deposit: 'deposit',
      credit: 'credit',
      insurance: 'insurance',
      online: 'online',
    },
  },
};
// ---------------------------------

// Footer
export const ContactsTel = {
  mobile: 'mobile',
  main: 'main',
};

export const SocialLinks = {
  facebook: 'facebook',
  instagram: 'instagram',
  twitter: 'twitter',
  youtube: 'youtube',
};
// ---------------------------------
