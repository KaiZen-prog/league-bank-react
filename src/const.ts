// Common
export const DESKTOP_MIN_WIDTH = 1024;
export const DIGIT_SPACE = 3;
export const PHONE_LENGTH = 18;
export const RATING_SCALE_MULTIPLIER = 20;

export const KeyCode = {
  ESC: 27,
};

export const AppRoute = {
  MAIN: '',
  CONVERTER: '/converter',
};
// ---------------------------------

// Converter
export const FLOAT_COEFFICIENT = 100;
export const MAX_HISTORY_LENGTH = 10;
export const MAX_DAYS = 8;

export const ConverterFormCurrencies = ['RUB', 'USD', 'EUR', 'KZT', 'GBP', 'CNY'];
export const ConverterCanvasCurrencies = ['USD', 'EUR', 'KZT', 'GBP', 'CNY'];

export const FormFields = {
  INPUT: 'currencyInput',
  OUTPUT: 'currencyOutput',
};

export const ConverterInputParams = {
  maxValue: 1000000,

  input: {
    id: 'currency-input',
    name: 'currencyInput',
  },

  output: {
    id: 'currency-output',
    name: 'currencyOutput',
  }
};
// ---------------------------------

// Calculator
export const REQUIRED_INCOME = 45;
export const QUANTITY_MONTH = 12;

export const CalculatorSteps = {
  params: 'params',
  request: 'request',
};

export const CreditPurpose = {
  none: {
    name: 'Выберите цель кредита',
    type: '',
  },
  mortgage: {
    name: 'Ипотечное кредитование',
    type: 'mortgage',
  },
  car: {
    name: 'Автомобильное кредитование',
    type: 'car',
  },
};

export const InputFields = {
  cost: 'cost',
  initialFee: 'initialFee',
  term: 'term',
};

export const LabelTypes = {
  car: 'car',
};

export const OfferTypes = {
  refusal: 'refusal',
};

export const SubmitButtonTypes = {
  preorder: 'preorder',
  request: 'request',
};

export const InputIconsTypes = {
  minus: 'minus',
  plus: 'plus',
};

export const InputTypes = {
  email: 'email',
  fullName: 'fullName',
  initialFee: 'initialFee',
  phone: 'phone',
  show: 'show',
  term: 'term',
  userInfo: 'userInfo',
};

export const MortgageParams = {
  type: 'mortgage',
  minCost: 1200000,
  maxCost: 25000000,
  costStep: 100000,
  minInitialFee: 10,
  minTerm: 5,
  maxTerm: 30,
  minCreditAmount: 500000,
  maternalCapitalValue: 470000,
  percent: {
    defaultPercent: 9.4,
    specialPercent: 8.5,
    amountForSpecialPercent: 15,
  },
};

export const CarParams = {
  type: 'car',
  minCost: 500000,
  maxCost: 5000000,
  costStep: 50000,
  minInitialFee: 20,
  minTerm: 1,
  maxTerm: 5,
  minCreditAmount: 200000,
  additionalToCar: {
    casco: 'casco',
    lifeInsurance: 'lifeInsurance',
  },
  percent: {
    defaultPercent: 16,
    specialPercent: 15,
    amountForSpecialPercent: 2000000,
    oneAddition: 8.5,
    allAdditions: 3.5,
  },
};
// ---------------------------------

// Slider
export const MainSlidesNames = {
  credit: 'credit',
  promo: 'promo',
  offices: 'offices'
};

export const MainSliderParams = {
  type: 'main-slider',
  isCarousel: true,
  slides: [
    {
      name: MainSlidesNames.credit,
      title: 'Лига Банк',
      slogan: 'Кредиты на любой случай',
      linkHref: 'calculator',
      link: 'Рассчитать кредит',
    },

    {
      name: MainSlidesNames.promo,
      title: 'Лига Банк',
      slogan: 'Ваша уверенность в завтрашнем дне',
      linkHref: '',
      link: '',
    },

    {
      name: MainSlidesNames.offices,
      title: 'Лига Банк',
      slogan: 'Всегда рядом',
      linkHref: 'map',
      link: 'Найти отделение',
    },
  ]
};

export const ServicesSlidesNames = {
  deposit: 'deposit',
  credit: 'credit',
  insurance: 'insurance',
  online: 'online'
};

export const ServicesSliderParams = {
  type: 'services-slider',
  isCarousel: false,
  slides: [
    {
      name: ServicesSlidesNames.deposit,
      tabName: 'Вклады',
      slogan: 'Вклады Лига Банка – это выгодная инвестиция в свое будущее',
      features: [
        'Проценты по вкладам до 7%',
        'Разнообразные условия',
        'Возможность ежемесячной капитализации или вывод процентов на банковскую карту',
      ],
      text: {},
      link: 'Узнать подробнее',
    },

    {
      name: ServicesSlidesNames.credit,
      tabName: 'Кредиты',
      slogan: 'Лига Банк выдает кредиты под любые цели',
      features: ['Ипотечный кредит', 'Автокредит', 'Потребительский кредит'],
      text: {
        firstLine: 'Рассчитайте ежемесячный платеж',
        secondLine: 'и ставку по кредиту воспользовавшись нашим ',
        link: 'кредитным калькулятором',
      },
      link: '',
    },

    {
      name: ServicesSlidesNames.insurance,
      tabName: 'Страхование',
      slogan: 'Лига Страхование — застрахуем все что захотите',
      features: [
        'Автомобильное страхование',
        'Страхование жизни и здоровья',
        'Страхование недвижимости',
      ],
      text: {},
      link: 'Узнать подробнее',
    },

    {
      name: ServicesSlidesNames.online,
      tabName: 'Онлайн-сервисы',
      slogan: 'Лига Банк — это огромное количество онлайн-сервисов для вашего удобства',
      features: [
        'Мобильный банк, который всегда под рукой',
        'Приложение Лига-проездной позволит вам оплачивать билеты по всему миру',
      ],
      text: {},
      link: 'Узнать подробнее',
    },
  ]
};
// ---------------------------------

// ReviewsForm
export const StarIconWidth = 37;

export const RatingParams = [
  {
    value: 5,
    id: 'star-5',
    title: 'perfect'
  },
  {
    value: 4,
    id: 'star-4',
    title: 'good'
  },
  {
    value: 3,
    id: 'star-3',
    title: 'not bad'
  },
  {
    value: 2,
    id: 'star-2',
    title: 'badly'
  },
  {
    value: 1,
    id: 'star-1',
    title: 'terribly'
  }
];
// ---------------------------------

// Footer
export const ContactsTel = {
  mobile: 'mobile',
  main: 'main',
};

export const SocialLinks = {
  facebook: 'facebook',
  instagram: 'instagram',
  twitter: 'twitter',
  youtube: 'youtube',
};
// ---------------------------------
