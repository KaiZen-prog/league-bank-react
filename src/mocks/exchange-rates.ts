export const exchangeRates = {
  '2024-01-19': {
    USD: 1,
    RUB: 88.573959,
    EUR: 0.918438,
    KZT: 452.633544,
    GBP: 0.787512,
    CNY: 7.1197
  },
  '2024-01-18': {
    USD: 1,
    RUB: 88.638497,
    EUR: 0.918649,
    KZT: 453.345273,
    GBP: 0.788835,
    CNY: 7.1263
  },
  '2024-01-17': {
    USD: 1,
    RUB: 88.105727,
    EUR: 0.919239,
    KZT: 453.139857,
    GBP: 0.791321,
    CNY: 7.1265
  },
  '2024-01-16': {
    USD: 1,
    RUB: 87.719298,
    EUR: 0.913662,
    KZT: 451.339299,
    GBP: 0.786436,
    CNY: 7.1073
  },
  '2024-01-15': {
    USD: 1,
    RUB: 88.043653,
    EUR: 0.913639,
    KZT: 451.160232,
    GBP: 0.785191,
    CNY: 7.111
  },
  '2024-01-14': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.160232,
    GBP: 0.78456,
    CNY: 7.1126
  },
  '2024-01-13': {
    USD: 1,
    RUB: 87.959997,
    EUR: 0.912076,
    KZT: 451.530905,
    GBP: 0.78456,
    CNY: 7.1106
  },
  '2024-01-12': {
    USD: 1,
    RUB: 88.495575,
    EUR: 0.910544,
    KZT: 451.995747,
    GBP: 0.782515,
    CNY: 7.09752
  }
};
