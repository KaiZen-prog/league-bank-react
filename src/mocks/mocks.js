export const promoSlides = [
  `img/desktop-slide-1.jpg`,
  `img/desktop-slide-2.jpg`,
  `img/desktop-slide-3.jpg`,
];
