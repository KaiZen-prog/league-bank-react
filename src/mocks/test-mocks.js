import moment from "moment";

export const InitialState = {
  CONVERSION_HISTORY: {
    conversionHistory: [
      {
        date: `18.09.2021`,
        currencyInput: {
          type: `USD`,
          amount: 6549
        },
        currencyOutput: {
          type: `EUR`,
          amount: 5585.07
        }
      },
      {
        date: `19.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 23
        },
        currencyOutput: {
          type: `RUB`,
          amount: 258.59
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 12458
        },
        currencyOutput: {
          type: `GBP`,
          amount: 10694.16
        }
      },
      {
        date: `13.09.2021`,
        currencyInput: {
          type: `CNY`,
          amount: 654
        },
        currencyOutput: {
          type: `RUB`,
          amount: 7363.93
        }
      },
      {
        date: `15.09.2021`,
        currencyInput: {
          type: `EUR`,
          amount: 105
        },
        currencyOutput: {
          type: `GBP`,
          amount: 89.6
        }
      },
      {
        date: `20.09.2021`,
        currencyInput: {
          type: `RUB`,
          amount: 105
        },
        currencyOutput: {
          type: `USD`,
          amount: 1.43
        }
      }
    ]
  },

  CONVERTER: {
    date: moment().utc().format(`YYYY-MM-DD`),
    exchangeRate: {
      USD: 1,
      RUB: 72.8797,
      EUR: 0.852813,
      GBP: 0.727776,
      CNY: 6.4662
    }
  }
};
