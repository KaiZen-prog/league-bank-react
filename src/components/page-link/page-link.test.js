import React from 'react';
import renderer from 'react-test-renderer';
import PageLink from './page-link';
import {AppRoute} from "../../const";
import {Router} from "react-router-dom";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {createMemoryHistory} from "history";
import {InitialState} from "../../mocks/test-mocks";

const mockStore = configureStore();
const history = createMemoryHistory({initialEntries: [`/`]});

test(`PageLink render correctly`, () => {
    const store = mockStore(InitialState);

    const tree = renderer
        .create(
            <Provider store={store}>
                <Router history={history}>
                    <PageLink
                        link={AppRoute.ROOT}
                        htmlClass={`some-class`}
                        description={`some description`}
                    />
                </Router>
            </Provider>
        ).toJSON();

    expect(tree).toMatchSnapshot();
});
