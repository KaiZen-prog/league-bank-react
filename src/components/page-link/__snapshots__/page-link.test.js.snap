// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLink render correctly 1`] = `
<a
  className="some-class"
  href="/"
  onClick={[Function]}
>
  some description
</a>
`;
